homeassistant/components/solarlog/const.py

```diff
@@ -31,146 +31,138 @@
-        json_key="yieldDAY",
         name="yield day",
         icon="mdi:solar-power",
         native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        factor=0.001,
     ),
     SolarLogSensorEntityDescription(
         key="yield_yesterday",
-        json_key="yieldYESTERDAY",
         name="yield yesterday",
         icon="mdi:solar-power",
         native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        factor=0.001,
     ),
     SolarLogSensorEntityDescription(
         key="yield_month",
-        json_key="yieldMONTH",
         name="yield month",
         icon="mdi:solar-power",
         native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        factor=0.001,
     ),
     SolarLogSensorEntityDescription(
         key="yield_year",
-        json_key="yieldYEAR",
         name="yield year",
         icon="mdi:solar-power",
         native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        factor=0.001,
     ),
     SolarLogSensorEntityDescription(
         key="yield_total",
-        json_key="yieldTOTAL",
         name="yield total",
         icon="mdi:solar-power",
         native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
         state_class=STATE_CLASS_TOTAL_INCREASING,
+        factor=0.001,
     ),
     SolarLogSensorEntityDescription(
         key="consumption_ac",
-        json_key="consumptionAC",
         name="consumption AC",
         native_unit_of_measurement=POWER_WATT,
         device_class=DEVICE_CLASS_POWER,
         state_class=STATE_CLASS_MEASUREMENT,
     ),
     SolarLogSensorEntityDescription(
         key="consumption_day",
-        json_key="consumptionDAY",
         name="consumption day",
         native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
         device_class=DEVICE_CLASS_ENERGY,
+        factor=0.001,
     ),
     SolarLogSensorEntityDescription(
         key="consumption_yesterday",
-        json_key="consumptionYESTERDAY",
         name="consumption yesterday",
         native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
         device_class=DEVICE_CLASS_ENERGY,
+        factor=0.001,
     ),
     SolarLogSensorEntityDescription(
         key="consumption_month",
-        json_key="consumptionMONTH",
         name="consumption month",
         native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
         device_class=DEVICE_CLASS_ENERGY,
+        factor=0.001,
     ),
     SolarLogSensorEntityDescription(
         key="consumption_year",
-        json_key="consumptionYEAR",
         name="consumption year",
         native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
         device_class=DEVICE_CLASS_ENERGY,
+        factor=0.001,
     ),
     SolarLogSensorEntityDescription(
         key="consumption_total",
-        json_key="consumptionTOTAL",
         name="consumption total",
         native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
         device_class=DEVICE_CLASS_ENERGY,
-        state_class=STATE_CLASS_TOTAL_INCREASING,
+        state_class=STATE_CLASS_MEASUREMENT,
```

Comment:
@emontnemery how are we handling net energy measurements now?

Reply:
I'll wait for the response of @emontnemery on this one. We are discussing this for the solarlog in this issue #55345 as well. The output of the Solarlog "total consumption" sensor is a total of "grid in" minus "grid out". It does not give us a split up in the json output (even when the device does measure it both). I was under the impression that STATE_CLASS_TOTAL_INCREASING should always be counting up (except for a reset of course).

Reply:
We should just drop the state class.

```suggestion
```

Reply:
Aren't the drops very little on this one, and we handle it automatically inside state class total increasing?

Reply:
It depends on the users installation, I think. On a sunny day it can go down a few kWh per day. Shouldn’t `total increasing` not be always counting up or can it deal with that?

and if we drop the state_class, can you still use it for the energy option in Home Assistant?

Reply:
To handle edge cases where the solar production includes consumption by the inverter system causing the amount of produced energy to decrease slightly during brief periods where the yield from the solar cells is less than is consumed by the converter, there's now a workaround in place where we don't trigger a new meter cycle if the decrease between two states is < 10%.

If it's an actual net sensor (grid in - grid out), it may also work, but I don't think it's a reliable solution long term.

Reply:
Just to make the discussion clear, the solarlog has two outputs, 
- `yield total`, which is the solar production (always increasing)
- `consumption total`, which is the sum of all connected consumption meters. Solarlog offers some consumption meter, including consumption meters that measure grid in and grid out. The json output of `consumption total` is only one value, grid in minus grid out.

The discussion is about `consumption total`, so indeed grid in minus grid out.

as it is the total, it might work, especially as the total increases. With low totals it may go wrong. Perhaps it is an idea to also allow a small decrease of x kWh per hour?

Reply:
It could work, but there are edge cases.
For example, if the sensor is off line for some time the difference between the last state and the first state update may be significant, thus incorrectly triggering a new meter cycle.

For now, I suggest to set this sensor to `state_class=STATE_CLASS_MEASUREMENT`, with a constant `last_reset`, `last_reset=homeassistant.util.dt.utc_from_timestamp(0)` if the total never resets.

Reply:
Ok, thanks. I've added `last_reset`.